### ISSUE TITLE
Solar Data GIS for Balkans

### ISSUE THREAD

**Issue by EgorBEremeev** [serbiancaseforspace/planet-balkan-hackathon#5]
Hi, all!

This idea is about to inspire people in usage of solar energy and reduce household fuel and electricity costs.
We can try do it by showing the comparison of local electricity prices and potential of Rooftop Solar installation.  
The one map can show this two type of data in real-time and\or historical average resolutions.

_Solar data_
There are two most used indicators about solar: Direct Normal Irradiance (DNI) and Global Horizontal Irradiance (GHI) layers.

> **Direct Normal Irradiance (DNI)** is direct beam radiation. Generally, it is used to calculate Concentrating PV (Photovoltaic Panels) output. Concentrating PV uses optics to concentrate the solar radiation in a way that captures only the DNI and not the diffuse or reflected components of the incident solar radiation.

> **Global Horizontal Irradiance (GHI)** is the total solar radiation incident on a horizontal surface. It is the sum of Direct Normal Irradiance (DNI), Diffuse Horizontal Irradiance, and ground-reflected radiation.

- [ ] the first question for an implementation is how Planet's platform may help to get data of these indicators? Should we looking for other data providers?

The electricity prices data probably should be free available by countries regulation agencies. However it may not be aggregated from different electricity providers.

- [ ] so, the second question for an implementation is finding Balkans-wide electricity historical and actual prices data

This is very brief idea description. Welcome to evaluate and contribute!

_About myself._
My personal goal for this hackaton is getting my very first experience with programming geo application and earth observation data processing pipelines. I'm familiar with python, google cloud platform (Functions, Storage, BigQuery), AWS (Storage, Image Rekognition), deep learning (TensorFlow, Keras). I got ARSET introduction training about remote-sensing, NASA Earth observation products, SAR (Sentinel 2) data processing.

**Comment by EgorBEremeev** [comment]
Hi, all!
This idea about inspire the use of solar energy and reduce household fuel and electricity costs.

We can try do it by showing the comparison of local electricity prices and potential of Rooftop Solar installation.  
The one map can show this two type of data in real-time and\or historical average resolutions.

_Solar data_
There are two most used indicators about solar: Direct Normal Irradiance (DNI) and Global Horizontal Irradiance (GHI) layers.

> **Direct Normal Irradiance (DNI)** is direct beam radiation. Generally, it is used to calculate Concentrating PV (Photovoltaic Panels) output. Concentrating PV uses optics to concentrate the solar radiation in a way that captures only the DNI and not the diffuse or reflected components of the incident solar radiation.

> **Global Horizontal Irradiance (GHI)** is the total solar radiation incident on a horizontal surface. It is the sum of Direct Normal Irradiance (DNI), Diffuse Horizontal Irradiance, and ground-reflected radiation.

- [ ] the first question for an implementation is how Planet's data services may help to get these indicators? Should we looking for other data providers?

The prices data probably should be free available. However it may not be aggregated from different electricity providers.

- [ ] the second question for an implementation find Balkans-wide electricity historical and actual price data


This is very brief idea description. Welcome to evaluate and contribute!

_About myself._
My personal goal for this hackaton is getting my very first experience with programming geo application and earth observation data processing pipelines. I'm familiar with python, google cloud platform (Functions, Storage, BigQuery), AWS (Storage, Image Rekognition), deep learning (TensorFlow, Keras). I got ARSET introduction training about remote-sensing, NASA Earth observation products, SAR (Sentinel 2) data processing.

**Comment by pajicf** [comment]
Hi Egor!

I like these projects which ought to inspire people and at the same time make it a bit easier to plan and calculate all the needed data for their green projects.

It reminds me a lot of EU JRC's [PVGIS](https://re.jrc.ec.europa.eu/pvg_tools/en/tools.html) (Photovoltaic Geographical Information System) tool which is focused on providing only the solar data and it also seems a bit too complicated tool to use for someone fresh in solar energy who is still just considering if solar panels are even an option and just wants to see how it compares to the traditional way of energy consumption.

The new tool you're proposing, to me, seems like a very good way to onboard new people into green energy space by providing them with actual consumption and prices data for their location while doing so in a more streamlined UX.
And because initially it will be focusing on the Balkans, it's going to be easier to get more accurate electricity prices and focus on less factors than the EU's PVGIS would.

For the solar data, if the Planets API can help that's going to be great, otherwise maybe the tool can make use of PVGIS existing API somehow?
 For the electricity prices, maybe [this](https://www.globalpetrolprices.com/electricity_prices/) service can help? They supposedly provide prices per kWh (households and businesses) for all Balkan countries and not sure, but it seems like you may query historical data through their API.

**Comment by EgorBEremeev** [comment]
Hi, Filip!

It is exactly my thoughts too to make clear and simple tool about potential of individual energetics.

These are the great data sources. PVGIS provides data about PV output potential in different type of installations. This is even more practically useful than direct solar radiation data.  If connected with https://www.globalpetrolprices.com/, it allows direct calculations of the cost saved.

It seems, good approach to focus on the segment of non-professional users who querying very basic question: "What is my individual profit of investing in green energetics?". They are real decision-makers and right tool can support them in the next steps. 

The tool could ask user to input some individual objective indicators, which are easy to assemble and easy to interpret for a person. Probably such indicator could be:
- actual [_monthly, over-winter, over-summer, else(?)_] [_costs, consumption_] for [_electricity, heating, fuel_]
- geopoint(s). 

- [ ] What factors about geopoint are reasonable to consider here, thinking about Balkans region? Probably, type of settlement [rural, small city, big city, else(?)], type of property [_individual house, block, office building, else(?)_]

Then, based on geopoint, the tool could get data from the PVGIS about potential output and combining with prices data calculate the potential of saved costs by solar energy. We can think what technology factors are reasonable to consider, like the type of installation

And as third and final step show infographics with comparison of actual consumption and saved costs.

What UI / UX approaches do you think are suitable for such a tool / application?

**Comment by pajicf** [comment]
Hi again Egor,

It seems like we're on the same page as for which kind of users should the initial product be focused on, beginner consumer segment in the Balkans just wanting to try out solar power and save some buck. That would cut out businesses, cut out people trying to go full solar and reduce the amount of data needed for the initial MVP and also allow us to make more assumptions for the user by which we would simplify the application.

For example, since the app would focus on consumers, we can assume that the type of the installation is residential and that they can easily install it on their property. Another factor would be how many sqm of installation can installed by the user which can be provided input or a average approximate calculation for homes in the Balkans.
There's also a solar panel type factor, should we calculate the installation cost for Monocrystalline, Polycrystalline or Thin-film. Again I think our advantage here would be defined user segment and location (Balkans), we can pick the solar panel type based on how mainstream is one region and also based on the assumption that the users are just trying to save a bit on already existing electrical grid, in which case we calculate the cheaper option. 

In short, as for the UX approach, I'd stick to the KISS (Keep it simple, stupid)  principle for the user, so that anyone can come with minimal knowledge and still be able to calculate and understand the savings that they can make by going solar.
Perhaps a good way to achieve this is through some kind of wizard which user goes through and at the end is presented with infographics and potentially a map of the Balkans similar to the PVGIS tool, but we should keep in mind that this is a 2 day hackathon and maybe we shouldn't go that much further yet.

Hope this makes sense and am interested of hearing your opinion on this approach...

**Comment by EgorBEremeev** [comment]
Well, the summary for weekend:
----

This idea is about to inspire people in usage of solar energy and reduce household fuel and electricity costs.

The new tool seems as a very good way to onboard new people into green energy space by providing them with actual consumption and prices data for their location while doing so in a more streamlined UX.

We can try do it by showing the comparison of local electricity prices and potential of Rooftop Solar installation.
The one app can show this two type of data in real-time and\or historical average resolutions.

**Product design**
**_Consumers_**
The initial product be focused on beginner consumer segment in the Balkans just wanting to try out solar power and save some buck.

**_Value_**
Anyone can come with minimal knowledge and still be able to calculate and understand the savings that they can make by going solar
 

_Assumptions_
1. the app would focus on consumers point of view
2. user will provide data about geolocation to filter and calculate PV indicators:

   - minimum city
   - max address
   - interactive point on map

    2.1. we can assume that the type of the installation is residential
    - that they can easily install it on their property.
     a) user can provide data about how many sqm of installation can installed by the user
    b) we can obtain average approximate calculation for homes in the Balkans.

    2.2. solar panel type factor, should we calculate the installation PV output for Monocrystalline, Polycrystalline or Thin-film
	  - we can pick the solar panel type based on how mainstream is one region
	  - or we can pick up the high-ouput type
	  - or we can pick up the lowest installation cost type

3. user will provide data about currect consumption in order to show comparison and potential savings:
	actual [monthly, over-winter, over-summer, else(?)] [costs, consumption] for electricity
	
	
**Hackathon MVP Scope suggestions**

**UX**
_Format:_
	- Web App
	- Landing Page size

_Concept_:
	- Calculator and\or
	- Data Story and\or
	- Education via Fact Sheets

_User input_: some kind of wizard
	- one page and\or
	- 2-3 step with input parameters

_App output_: Infographics, Data Story
	- 1-3 indicators
	- user location area map
	- geospatial
	
**Architecture**
_Front end_: 
 -web app

_Back end_:
	- Connectors to Data Sources API
		-- PV output
			--- [PVGIS](https://ec.europa.eu/jrc/en/pvgis)
		-- Electricity prices
			--- https://www.globalpetrolprices.com/
			--- https://data.europa.eu/euodp/en/data/dataset/O0KGQ8bifnbvc85xZncGA
	- Calculations of aggregated indicators
	- Map layers pixel calculation and rendering

_Tools_:
	- [Google App Engine](https://cloud.google.com/appengine/docs) web app
	- [Google Earth Engine](https://earthengine.google.org/).
	- https://developers.google.com/earth-engine/guides
